src/Expression/Simplifier.purs

```diff
@@ -0,0 +1,70 @@
+module Expression.Simplifier where
+
+import Prelude
+
+import Control.Alt ((<|>))
+import Data.Maybe (Maybe(..), fromMaybe)
+import Expression.Syntax (BinaryOperation(..), Expression(..), UnaryOperation(..))
+import Math (pow)
+
+simplify :: Expression -> Expression
+simplify (ExpressionUnary operation expression) = case simplify expression, operation of
+  ExpressionLiteral value, Neg -> ExpressionLiteral (-value)
+  simplifiedExpression, _ -> ExpressionUnary operation simplifiedExpression
+
+simplify (ExpressionBinary Power leftExpression (ExpressionLiteral 1.0)) = simplify leftExpression
+
+simplify (ExpressionBinary Power _ (ExpressionLiteral 0.0)) = ExpressionLiteral 1.0
+
+simplify (ExpressionBinary Power (ExpressionVariable "e") rightExpression) = ExpressionUnary Exp $ simplify rightExpression
+
+simplify (ExpressionBinary operation leftExpression rightExpression) =
+  fromMaybe (ExpressionBinary operation simplifiedLeftExpression simplifiedRightExpression)
+    $ trimSimpleNodes simplifiedLeftExpression simplifiedRightExpression operation
+    <|> trimConstantLeafNodes simplifiedLeftExpression simplifiedRightExpression operation
+    <|> trimTimesOperations simplifiedLeftExpression simplifiedRightExpression operation
+    <|> trimPlusOperations simplifiedLeftExpression simplifiedRightExpression operation
+  where
```

Comment:
I like this way of breaking the simplification down and chaining back.
Also, well done on completing so many useful simplifications.

Reply:
Thank you